File: annotation-processors/src/main/java/com/comsysto/livingdoc/annotation/processors/plantuml/model/SequenceDiagram.java
package com.comsysto.livingdoc.annotation.processors.plantuml.model;

import static java.util.stream.Collectors.toSet;

import lombok.Value;

import java.util.List;
import java.util.Set;

@Value
public class SequenceDiagram {

    /**
     * the diagram title.
     */
    String title;

    /**
     * Any PlantUml files to be included.
     */
    List<String> includeFiles;

    /**
     * The executable parts to be rendered in this diagram.
     */
    List<ExecutablePart> parts;

    public Set<String> typeNames() {
        return parts.stream()
            .map(part -> part.getAnnotated().getEnclosingElement().asType())
            .map(ClassDiagram::simpleTypeName).collect(toSet());
    }
}

File: annotation-processors-kotlin/src/main/kotlin/model/SequenceDiagram.kt
package com.comsysto.livingdoc.kotlin.annotation.processors.model

import model.ExecutablePart

class SequenceDiagram(

        /**
         * the diagram title.
         */
        title: String? = null,

        /**
         * Any PlantUml files to be included.
         */
        includeFiles: List<String> = listOf(),

        val parts: List<ExecutablePart>

) : Diagram(title, includeFiles)